### PR TITLE
Set git branch to empty string instead of None

### PR DESCRIPTION
**Pull request by vrutkovs**
This unblocks builds using rhpkg, which doesn't pass git branch parameter

### CODE REVIEW

**Comment by twaugh** [comment]
I think it should default to 'unknown' instead of the empty string, and the equivalent code in osbs/utils.py:make_name_from_git() should be removed.

Is testing this hard?

**Comment by vrutkovs** [comment]
Sure, sounds like a good idea to me. Testing it is fairly easy on Jenkins.

I wonder if using something like ```git branch --contains <commit id>``` is a good idea. What do you guys think?

**Comment by twaugh** [comment]
We looked at this a while ago, and there's basically no way to infer the branch name in the general case. For example: imagine when two branches are at the same commit.

**Comment by vrutkovs** [comment]
Yes, this and introduction of python git bindings are my biggest concerns.

I'll set this to 'unknown' for now

File: osbs/utils.py [hunk 1]
```diff
@@ -311,7 +311,7 @@ def make_name_from_git(repo, branch, limit=53, separator='-'):
     """
 
     repo = git_repo_humanish_part_from_uri(repo)
-    branch = branch or 'unknown'
+    branch = branch
```
Comment (twaugh) [inline osbs/utils.py]: no-op :-)

**Comment by twaugh** [comment]
FWIW, if we can't get it working so that triggers are removed correctly when no branch is passed in, and easy alternative fix would be to do the equivalent default 'unknown' substitution when setting the annotation.

**Comment by vrutkovs** [comment]
Right, moved this part to the ```set_label``` part, otherwise autorebuild is being triggered.

Tested on local koji instance

**Comment by twaugh** [comment]
Can we make the test_render_prod_request_trigger_missing_param() test-case verify that the git-branch label is never set to None?

**Comment by twaugh** [comment]
LGTM

**Comment by lcarva** [comment]
I wonder if `set_label` should be modified to never allow `None` values.

**Comment by twaugh** [comment]
Couldn't hurt.

**Comment by twaugh** [comment]
LGTM

**Comment by lcarva** [comment]
LGTM

**Comment by mmilata** [comment]
LGTM

**Pull request merged**